### ISSUE TITLE
There is no default font selected in the Preference window

### ISSUE THREAD

**Issue by karim** [sqlitebrowser/sqlitebrowser#749]
### Details for the issue

DB4S has a default font that it is using for the "Browse Data" tab. You can see it in the next image.

![db4s_1](https://cloud.githubusercontent.com/assets/8761081/18008694/6887ef3c-6ba9-11e6-87dc-b8876c7d2cff.jpg)

However, if you open the "Preference" window to change anything, or simply opened it and closed it without changing anything, it will change the default font and make it look like this..

![db4s_2](https://cloud.githubusercontent.com/assets/8761081/18008733/a15a8298-6ba9-11e6-91c7-92a5bbc72a9d.jpg)

This only happens if you click OK, whether you changed settings or not, but not if you click Cancel.

In both cases, there is no font selected in the "Preference" window. It is empty like this..

![db4s_3](https://cloud.githubusercontent.com/assets/8761081/18008945/71a00748-6baa-11e6-8ba2-c8531b220222.jpg)

To go back to the font I want in the first image, I have to edit the registry and delete `font` value from `HKCU\Software\sqlitebrowser\sqlitebrowser\databrowser` key, then restart DB4S.

What is the default font that is used in the first image? I think this will be fixed if I set it manually.

### Useful extra information

I think this is related to #589.

#### I'm opening this issue because:

- [x] DB4S has a bug

#### I'm using DB4S on:

- [x] Windows 7 SP1 (64-bit)

#### I'm using DB4S version:

- [x] 3.9.0

#### I have also:

- [ ] Tried out the nightly version: https://github.com/sqlitebrowser/sqlitebrowser#windows
- [x] Searched for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
@karim Thanks for pointing this out.  This is happening here too on a freshly installed system.

It sounds like we need to adjust how the default font values are chosen and displayed in Preferences, so that it's not blank/incorrect like in your screenshots.

**Comment by innermous** [comment]
@karim is `font` value empty in your .conf file, after you click OK? It's just `font=` right?

**Comment by karim** [comment]
@innermous Yes.

![db4s_4](https://cloud.githubusercontent.com/assets/8761081/18013302/b7fc91ee-6bbd-11e6-9eec-ea17e11115dd.jpg)

There is no .conf file. Everything is saved in Windows Registry.

**Comment by innermous** [comment]
@karim if I send a fix in next few minutes(via PR), are you able to test it?

**Comment by karim** [comment]
@innermous Only if you are going to build it, cause I don't know how. :smile:

**Comment by innermous** [comment]
@karim I am not a windows user...
@justinclift says he can reproduce it, so I hope he will test changes.

**Comment by justinclift** [comment]
@innermous Yep, I can test it here. :smile:

I can also build a package for @karim to test with on Windows too.  One of my VM snapshots has everything set up for building, I just need to give it updated source code. 😀

**Comment by karim** [comment]
@justinclift Let me know when you build it so I can test.

**Comment by justinclift** [comment]
Building it now.  Gimme 15+ minutes, as the build server isn't super fast. :wink:

**Comment by justinclift** [comment]
@karim As a side note, if you're ever interested in building it yourself, we added step-by-step instructions to set up a development environment here:

&nbsp; &nbsp; https://github.com/sqlitebrowser/sqlitebrowser/wiki/Setting-up-a-Win64-development-environment-for-DB4S

It's not tricky at all... just very time consuming the first time around. 😼

**Comment by karim** [comment]
@justinclift Thanks. :+1: I saved the link and I will give it a try later.

**Comment by justinclift** [comment]
@karim Try this one out: :smile:

* http://nightlies.sqlitebrowser.org/osx-onceoffs/DB%20Browser%20for%20SQLite-3.9.0-issue749v1-win64.exe

It's the 3.9.0 release +@innermous's potential fix.

**Comment by justinclift** [comment]
@innermous Yep, that fixed the initial blank entry for the Font in the "Data Browser" tab, so the concept seems good. :smile:

The same adjustment needs to be done for the "SQL editor font" in the "SQL" tab too.

**Comment by innermous** [comment]
Fixed [here](https://github.com/sqlitebrowser/sqlitebrowser/commit/b15c7f20a5c735f6e059d6e5a7a91df435a3526a).

**Comment by justinclift** [comment]
We need to get the other field fixed too. :wink:

**Comment by karim** [comment]
Sorry for the delay, and.. yes, it works! :tada:

It is no longer blank and it is now set to "MS Shell Dlg 2".

Also, and like @justinclift said, we need to change the font for the "Execute SQL" tab too.

**Comment by justinclift** [comment]
@karim We can probably get this properly fixed (eg with the Execute SQL one too) in the next day or so.

As a thought, if you use the binary editing mode in the "Edit Database Cell" panel much, try setting the "Data Browser" Font field to Courier.  Or any other fixed width font really.

The binary editing mode seems to look weird using a font that's not fixed width, at least on my system. :wink:

**Comment by karim** [comment]
@justinclift I don't use the "Edit Database Cell", I edit fields directly which is very fast. I move around with the arrow keys and start typing on a highlighted field to edit. It is one of the things I like a lot about DB4S.

Thanks for making me a working build, and thanks to @innermous for making a quick fix.

**Comment by justinclift** [comment]
No worries at all.  I use keyboard navigation a lot too when editing text data. :smile:

**Comment by justinclift** [comment]
@karim If you have a few minutes, would you be ok to try this build out?

* http://nightlies.sqlitebrowser.org/win64-onceoffs/DB%20Browser%20for%20SQLite-3.9.0-issue749v2-win64.exe

In theory, that should be fixed for both of the font fields. :smile:

**Comment by karim** [comment]
@justinclift Yes, I see now that the font in "Execute SQL" tab has changed. It is much better now, like it used to be. This font doesn't change when you change `font` in "Preferences" window, right?

**Comment by justinclift** [comment]
Hmmm, from memory, it should be controllable through the Preferences.  I **think** it's the "SQL editor font" line in the "SQL" tab, but don't quote me on that.

I'd have to look at the code to be sure, and I've closed down that VM.

If it's important to you though, I can look over it in the morning to refresh my memory?

**Comment by karim** [comment]
Ahhhh, yes. :smile: I see it now. It is like you said in the "SQL" tab in "Preferences".

No no, it is not important at all. I was just wondering, that's all. Cause I changed the font like I normally do in "Data Browser" and it was changed, but then I switched to "Execute SQL" and it was the same, so I thought something was wrong. I didn't notice it has its own font settings in "SQL" tab, sorry.

I tested both (changing font in Data Browser tab and Execute SQL tab) and they are working very good. Thanks a lot for all your help. :+1:
